### ISSUE TITLE
BUG:  Horizontal scroll issue

### ISSUE THREAD

**Issue by TheTushar2696** [hoangnm/react-native-week-view#243]
**Describe the bug**
On horizontal scroll, it becomes stuck sometimes and then resumes to scroll again, attaching a video of the same

https://user-images.githubusercontent.com/40525668/180362825-0c2e21ab-a77f-4839-9e7e-167f5ec7f863.mov

**Expected behavior**
Should scroll smoothly

**Comment by pdpino** [comment]
This is a trade-off issue between loading less data (UI responding faster, but possible blank spaces) vs loading more data (UI slower, less blank spaces):

* When the view is near the end to the right the component loads and render more weeks to the right. This process takes time
* If the user scrolls too often/fast, you see the behavior you're describing
* We currently try to optimize this with some of [these FlatList props](https://reactnative.dev/docs/optimizing-flatlist-configuration). You can check the props pros and cons for the trade-off. For example we set `windowSize=5` (5 weeks = 2 to the left + 1 central + 2 to the right)

Idea: we could expose props `windowSize`, `updateCellsBatchingPeriod`, `maxToRenderPerBatch`, so the user can tune them better. What do you think?

**Comment by TheTushar2696** [comment]
@pdpino  Just thinking out loud, If we give a window size as you mentioned, will it work well with goToDate function, because I've observed a similar behaviour when the user is moving to a specific date.

**Comment by pdpino** [comment]
> will it work well with goToDate

you are right, it won't work well. See this similar issue: #207 when navigating to a far date (or even to close dates)

We are trying to solve that separately from the flatlist optimizations. The process when swiping with the finger (moving one page by one) is slightly different than goToDate (navigating to an arbitrary date)

**Comment by TheTushar2696** [comment]
@ Got it, Sure then we can try to add widow size, Since you mentioned we already are discussing this already, are you planning to take this up?

**Comment by pdpino** [comment]
@TheTushar2696 I sent a PR exposing the props. You can try different values to get better results, but I cannot assure you there will be _better_ results

Some other comments:
1. I'm implementing other optimizations to be able to implement some other features. These optimizations could help
2. I think a better solution would be to block the scrolling for the user while the component is loading. I want to add this, but I need to do (1) first

**Comment by TheTushar2696** [comment]
Hi @pdpino,
Got it, Also one more thing I noticed is when I am populating large number of events it is showing this in console, any idea about it?

<img width="394" alt="Screenshot 2022-08-03 at 2 52 15 AM" src="https://user-images.githubusercontent.com/40525668/182475699-796528f1-fceb-4983-9f25-829007ca6f5e.png">

**Comment by pdpino** [comment]
@TheTushar2696 I haven't seen that one with week-view

[This](https://stackoverflow.com/a/60234461/9951939) mentions passing large objects to console log
